### PR TITLE
Improving readability of the code base

### PR DESCRIPTION
**Pull request by AntonioConsiglio**
arguments function type have been added.

### CODE REVIEW

File: src/ml.py [hunk 1]
```diff
@@ -5,9 +5,19 @@
...
+from sklearn.ensemble import RandomForestClassifier
 
+from typing import Union,Tuple,List,Optional
 
-def train_evaluate_model(x_train, y_train, x_test, y_test, model, verbose):
+def train_evaluate_model(x_train:pd.DataFrame, 
+                         y_train:pd.DataFrame, 
+                         x_test:pd.DataFrame,
+                         y_test:pd.DataFrame, 
+                         model:Union[Lasso,RandomForestClassifier],
```
Comment (apalladi) [inline src/ml.py:17]: Is there a way to specify it in a more general way? Lasso and RandomForestClassifier are not the only models supported (although they are the only ones in the example and in the tests, for the moment)

File: src/ml.py [hunk 2]
```diff
@@ -40,7 +50,8 @@ def train_evaluate_model(x_train, y_train, x_test, y_test, model, verbose):
     return model
 
 
-def get_feature_importances(trained_model, column_names):
+def get_feature_importances(trained_model:Union[Lasso,RandomForestClassifier],
```
Comment (apalladi) [inline src/ml.py]: same comment as above

File: src/ml.py (hunk 1 above)
Comment (apalladi) [inline src/ml.py]: Is there a way to specify it in a more general way? Lasso and RandomForestClassifier are not the only models supported (although they are the only ones in the example and in the tests, for the moment)

File: src/ml.py [hunk 3]
```diff
@@ -316,7 +335,7 @@ def get_relevant_features(
 
         epoch += 1
 
-    if isinstance(filename_output, str):
+    if filename_output is not None:
```
Comment (apalladi) [inline src/ml.py:334]: Much better, I agree!

**Review by apalladi — COMMENTED** [review]
Thank you Antonio for your commit. It improves a lot the readability. I would like to merge it as soon as possible, just have a look at my comments when you have time.

File: src/ml.py (hunk 1 above)
Comment (AntonioConsiglio) [reply src/ml.py]: image

We can do like scikit-learn. (Base Estimator)

**Review by apalladi — APPROVED** [review]

**Pull request merged**